src/app/modules/item/task/task-xd-pr.ts

```diff
@@ -0,0 +1,303 @@
+ */
+export class Task {
+  platformSet = false;;
+  chan: RxMessagingChannel;
+
+  constructor(chan: RxMessagingChannel) {
+    this.chan = chan;
+  }
+
+  destroy(): void {
+    this.chan.destroy();
+  }
+
+  setPlatform(platform : Platform): void {
+    if (this.platformSet) {
+      throw new Error("Task already has a platform set");
+    }
+    this.chan.bind('platform.validate', (mode: string) => platform.validate(mode));
+    this.chan.bind('platform.getTaskParams', (keyDefault? : [string, TaskParamsValue]) => platform.getTaskParams(keyDefault));
+    this.chan.bind('platform.showView', (view : TaskView) => platform.showView(view));
+    this.chan.bind('platform.askHint', (hintToken : string) => platform.askHint(hintToken));
+    this.chan.bind('platform.updateDisplay', (data : TaskDisplayData) => platform.updateDisplay(data));
+    this.chan.bind('platform.openUrl', (url : string) => platform.openUrl(url));
+    this.chan.bind('platform.log', (data : TaskLog) => platform.log(data));
+
+    // Legacy calls
+    this.chan.bind('platform.updateHeight', (height : number) => platform.updateDisplay({ height: height }));
+  }
+
+  /**
+   * Task API functions
+   */
+  load(views : Object): Observable<void> {
+    return this.chan.call({
+      method: "task.load",
+      params: views,
+    });
+  }
+
+  unload() : Observable<void> {
+    return this.chan.call({
+      method: "task.unload",
+      timeout: 2000,
+    });
+  }
+
+  getHeight() : Observable<number> {
+    // TODO: validator
+    return this.chan.call({
+      method: "task.getHeight",
+      timeout: 500,
+    });
+  }
+
+  updateToken(token : string) : Observable<void> {
+    return this.chan.call({
+      method: "task.updateToken",
+      params: token,
+      timeout: 10000,
+    });
+  }
+
+  getMetaData() : Observable<TaskMetaData> {
+    // TODO: validator
+    return this.chan.call({
+      method: "task.getMetaData",
+      timeout: 2000
+    });
+  }
+
+  getAnswer() : Observable<string> {
+    // TODO: validator
+    return this.chan.call({
+      method: "task.getAnswer",
+      timeout: 2000
+    });
+  }
+
+  reloadAnswer(answer : string) : Observable<void> {
+    return this.chan.call({
+      method: "task.reloadAnswer",
+      params: answer,
+      timeout: 2000
+    });
+  }
+
+  getState() : Observable<string> {
+    // TODO: validator
+    return this.chan.call({
+      method: "task.getState",
+      timeout: 2000
+    });
+  }
+
+  reloadState(state : string) : Observable<void> {
+    return this.chan.call({
+      method: "task.reloadState",
+      params: state,
+      timeout: 2000
+    });
+  }
+
+  getViews() : Observable<TaskViews> {
+    // TODO: validator
+    return this.chan.call({
+      method: "task.getViews",
+      timeout: 2000
+    });
+  }
+
+  showViews(views : Object) : Observable<void> {
+    return this.chan.call({
+      method: "task.showViews",
+      params: views,
+      timeout: 2000
+    });
+  }
```

Comment:
I find quite confusing to have `showViews` here and in the "listeners`. I think they are two different things but I am not 100% sure. If they are different, probably they should have a different name.

Reply:
As it's a function defined in the API, I can't really rename it. They're not really different, as the calls are basically meant to synchronize the frontend and the task as to what should be displayed (task statement, editor, hints, ...). Most of the time, the platform will tell the task "display the editor now", but sometimes the task can change view and hence needs to tell the platform "I'm actually displaying the hints now".

The tabs in the frontend, above the task, will allow to select the task view to be displayed.

Reply:
But without changing the "method" name in the API the function can have a more explicit name without problem, right? 
I wonder if it wouldn't make clearer to have the TS functions named "showViewsInTask" / "showViewsInPlatform"

Reply:
I renamed the functions to `showViewsInTask` and `viewsShownByTask` ; for the second name, I think it indicates better what happens (some views were just shown by the task, which then communicates it to the platform).

In the next PR (item-display modifications), many of these functions will actually be used in a proper way, and I'll take the chance to add comments describing their use and behavior. Moreover the "platform" functions will be moved to item-display.

Reply:
Ah yes `viewsShownByTask` is better, actually I was wrong about I think it was doing.